src/compiler/checker.ts

```diff
@@ -10547,62 +10547,102 @@ module ts {
             return false;
         }
 
-        function getSymbolsInScope(location: Node, meaning: SymbolFlags): Symbol[] {
+        function getSymbolsInScope(location: Node, meaning: SymbolFlags, predicate?: (symbol: Symbol) => boolean): Symbol[] {

```

Comment:
I only measured after the fix.  The time to get the details was in the low number of MS.  Effectively, all the time was spent in the Roslyn delay timer, as well as just presenting the item.  Computing of the item was effectively nothing.  (This wasn't true for "Diagnostics".  but that's a known issue where we end up building up a massive display string for that special type).